### ISSUE TITLE
[UX] Add checks to make sure that users do not come across WTFs while configuring text formats.

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#1033]
In https://github.com/backdrop/backdrop-issues/issues/101#issuecomment-114659010, @quicksketch said:

> You may need to order the filters so that the "convert image captions" filter comes after the HTML filter, either that or add ```<figure>``` and ```<figcaption>``` to the HTML filter tag list.

Can we have these (and other similar conditions) trigger some inline warning for the user? I was thinking something like this (hacked HTML code mock-up - not an actual thing):

![text_format_config-conditional_hints](https://cloud.githubusercontent.com/assets/2423362/8319617/37e2e2e4-1a54-11e5-92f8-1cb285924221.png)

The red color + the yellow background should be enough to draw the user's attention to the matter.

**Comment by docwilmot** [comment]
:+1:

**Comment by klonos** [comment]
Maybe "after" is too vague and should better be "below" instead, but anyways this is about the concept and not the exact wording we'll use.

**Comment by MrHaroldA** [comment]
Shouldn't the warning be placed in the "Limit allowed tags" filter description? Other filters have possibly no notion of each other, especially since you can install extra filters that may be affected too.

**Comment by klonos** [comment]
Yes, in this specific case both warnings could fire at the same time for the respective filter. The "Move filter below limit HTML tags filter." for the *Convert image captions* filter + the "Add <figure> and <figcaption> to the list of allowed HTML tags." for the *Limit allowed HTML tags* filter. A check could fire each time a filter config changes + each time the filter order changes. In this specific case, either of the changes (adding the tags or re-ordering the filters appropriately) would satisfy the conditions and both warnings would disappear at the same time.

**Comment by MrHaroldA** [comment]
What if a filter could define it's required (or optional) html tags? The limit filter could check that and spit out specific errors/warnings...

**Comment by docwilmot** [comment]
> What if a filter could define it's required (or optional) html tags? The limit filter could check that and spit out specific errors/warnings...

That would be great; the filters could then just autoarrange and set a message saying so. Perhaps a checkbox to override the autoarranging.

**Comment by klonos** [comment]
Yes! Great idea :+1:
